### ISSUE TITLE
Не качает ресурсы с FastDL на некоторых серверах

### ISSUE THREAD

**Issue by Vladislav4KZ** [FWGS/xash3d-fwgs#1838]
### Проблема проявляется на данных серверах:
**Infotex[Experimental Weapon Lab]** - ``94.181.174.35:27018``
**MaddyMurk Deathmatch** - ``hl.maddymurk.ru:27015``
Оба сервера работают на движке GoldSource.

В консоли появляются ошибки такого рода:
```
Error: sound/city_scope/ba_close.wav: bad response: HTTP/1.1 301 Moved Permanently
Error: no servers to download sound/city_scope/ba_close.wav!
```

Проверял на:
Xash3D FWGS v49/0.21
Номер сборки: 3481
Тип сборки: win32-i386
Коммит: https://github.com/FWGS/xash3d-fwgs/commit/10bfadb556b9043f4dd1b4672d2342485be0e457
Ветка: master

ОС: Windows 10 Pro x64 22H2
Номер сборки ОС: 19045.5011

**Comment by a1batross** [comment]
О, пора редирект реализовывать.

**Comment by a1batross** [comment]
Тут всё-таки не cloudflare, а обычный редирект на себя же с HTTPS.

```
$ curl -v http://ra4fhe.narod.ru/sprites/weapon_skull1.txt
* Host ra4fhe.narod.ru:80 was resolved.
* IPv6: (none)
* IPv4: 193.109.247.233
*   Trying 193.109.247.233:80...
* Connected to ra4fhe.narod.ru (193.109.247.233) port 80
> GET /sprites/weapon_skull1.txt HTTP/1.1
> Host: ra4fhe.narod.ru
> User-Agent: curl/8.9.1
> Accept: */*
> 
* Request completely sent off
< HTTP/1.1 301 Moved Permanently
< Server: nginx
< Date: Mon, 14 Oct 2024 04:10:23 GMT
< Content-Type: text/html
< Content-Length: 178
< Connection: keep-alive
< Keep-Alive: timeout=15
< Location: https://ra4fhe.narod.ru/sprites/weapon_skull1.txt
< X-Frame-Options: SAMEORIGIN
< 
<html>
<head><title>301 Moved Permanently</title></head>
<body bgcolor="white">
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx</center>
</body>
</html>
* Connection #0 to host ra4fhe.narod.ru left intact
```

Попробовал подсоединиться к hl.maddymurk.ru:27015 с отключенным HTTP: файл оно конечно качает, но очень медленно. И это сжатые фрагменты.

Возможно, нам действительно нужен тут HTTPS. Насколько я знаю, голдсорс сейчас при скачивании с FastDL использует SteamHTTP из SteamAPI и судя по [документации](https://partner.steamgames.com/doc/api/isteamhttp) он поддерживает HTTPS.

**Comment by nekonomicon** [comment]
SSL доступен в любом случае везде, на худой конец можно mbedTLS или bearSSL прикрутить.

**Comment by SNMetamorph** [comment]
> SSL доступен в любом случае везде, на худой конец можно mbedTLS или bearSSL прикрутить.

Да, надо, и наверное всё же желательно сертификаты тащить вместе с движком, ибо у нас есть платформы, где их нет, либо они безнадёжно устарели (PS Vita, например).

**Comment by a1batross** [comment]
Проблемы с некоторыми HTTP серверами я пофиксил.

Некоторым серверам нужен `Accept: */*`, а так же нужен слеш в начале пути в запросе.
